### DISCUSSION TITLE
Dependency question

### DISCUSSION THREAD

**Discussion by Pavel-Sachivkin** [viewflow/viewflow#391] [General]
Hello, can you please tell me if your library can be used without using celery and redis and other similar libraries? Thank you!

**Comment by kmmbvnr** [comment]
Certainly!

Viewflow is a component-based workflow library. Its design allows you to use only the components that fit your specific needs. If you do not involve the components that require Celery, Redis, or other similar libraries, then there is no need to integrate or use them.

For workflows that need concurrency control to avoid race conditions, Viewflow provides the lock.select_for_update lock mechanism. This utilizes PostgreSQL's native row-level locking capability, ensuring that only one transaction can modify the data at a time. This means that even without incorporating additional libraries like Celery and Redis, you can still maintain data consistency and reliability in your workflows using Viewflow combined with a PostgreSQL backend.

**Comment by Pavel-Sachivkin** [comment]
Thank you very much!
